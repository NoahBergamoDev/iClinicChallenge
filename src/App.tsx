import React from 'react'
import MainNavigator from './navigation/MainNavigator'
import Toast from 'react-native-toast-message'

const App = () => {
    return (
        <>
            <MainNavigator />
            <Toast ref={ref => Toast.setRef(ref)} visibilityTime={5000} />
        </>
    )
}

export default App
