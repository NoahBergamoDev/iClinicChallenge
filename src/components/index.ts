import AutocompleteInput from './autocompleteInput/AutocompleteInput'
import Button from './button/Button'
import Title from './title/Title'
import Input from './input/Input'

export { AutocompleteInput, Button, Title, Input }
