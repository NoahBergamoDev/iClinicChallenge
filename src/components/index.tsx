import * as Button from './button/Button';

export { Button };
