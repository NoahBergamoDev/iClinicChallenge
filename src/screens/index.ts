import AddEditPrescriptionScreen from './AddEditPrescriptionScreen/AddEditPrescriptionScreen'
import HomeScreen from './home/HomeScreen'
import LoginScreen from './login/LoginScreen'
import PrescriptionDetailScreen from './prescriptionDetail/PrescriptionDetailScreen'
import PrescriptionListScreen from './precriptionList/PrescriptionListScreen'
export {
    AddEditPrescriptionScreen,
    HomeScreen,
    LoginScreen,
    PrescriptionDetailScreen,
    PrescriptionListScreen,
}
