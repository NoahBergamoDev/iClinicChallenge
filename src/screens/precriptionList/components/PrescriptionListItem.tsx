import React, { FC } from 'react'
import { Text, TouchableOpacity } from 'react-native'
import { colors } from '../../../utils/'
import { Prescription } from '../../../utils/types/Types'

interface Props {
    prescription: Prescription
    onPress: () => void
}

const PrescriptionListItem: FC<Props> = props => {
    const {
        onPress,
        prescription: { patient, physician },
    } = props
    return (
        <TouchableOpacity
            style={{
                flex: 1,
                backgroundColor: 'white',
                marginVertical: 8,
                padding: 20,
                borderRadius: 8,
            }}
            onPress={onPress}
        >
            <Text
                style={{
                    fontSize: 20,
                    color: colors.BLACK,
                    fontWeight: 'bold',
                    marginBottom: 8,
                }}
            >
                {patient.name}
            </Text>
            <Text
                style={{
                    fontSize: 16,
                    color: colors.MEDIUM_GRAY,
                    marginBottom: 4,
                }}
            >
                {physician.name}
            </Text>
            <Text
                style={{
                    fontSize: 16,
                    color: colors.MEDIUM_GRAY,
                }}
            >
                <Text style={{ fontWeight: 'bold' }}>CRM: </Text>
                {physician?.crm?.split('-', 2)[1]}
            </Text>
        </TouchableOpacity>
    )
}

export default PrescriptionListItem
