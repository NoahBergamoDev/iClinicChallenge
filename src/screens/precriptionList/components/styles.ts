import styled from 'styled-components/native'
import { colors } from '../../../utils'

export const Container = styled.TouchableOpacity`
    background-color: ${colors.WHITE};
    margin: 8px 0 8px 0;
    padding: 20px;
    border-radius: 8px;
`

export const PatientNameText = styled.Text`
    font-size: 20px;
    color: ${colors.BLACK};
    font-weight: bold;
    margin-bottom: 8px;
`
export const PhysicianNameText = styled.Text`
    font-size: 16px;
    color: ${colors.MEDIUM_GRAY};
    margin-bottom: 4px;
`

export const CRMText = styled.Text`
    font-size: 16px;
    color: ${colors.MEDIUM_GRAY};
`

export const BoldText = styled.Text`
    font-weight: bold;
`
