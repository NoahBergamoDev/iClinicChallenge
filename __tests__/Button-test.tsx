import React from 'react'
import renderer from 'react-test-renderer'
import { Button } from '../src/components'

const props = {
    label: 'Button Text',
    onPress: () => {},
}
test('renders correctly', () => {
    const tree = renderer
        .create(<Button label={props.label} onPress={props.onPress} />)
        .toJSON()
    expect(tree).toMatchSnapshot()
})
