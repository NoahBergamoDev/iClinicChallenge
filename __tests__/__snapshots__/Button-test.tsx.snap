// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#44A4E3",
      "borderRadius": 8,
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 16,
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
      "width": "90%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
        },
      ]
    }
  >
    Button Text
  </Text>
</View>
`;
